recipes/Voicebank/enhance/SGMSE/train.py

```diff
@@ -0,0 +1,874 @@
+import argparse
+import os
+from datetime import datetime
+from pathlib import Path
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from hyperpyyaml import load_hyperpyyaml
+from pesq import pesq
+from sgmse.util.other import pad_spec
+from torch.utils.tensorboard import SummaryWriter
+from torchmetrics.functional.audio import (
+    scale_invariant_signal_distortion_ratio as si_sdr,
+)
+from torchmetrics.functional.audio import (
+    short_time_objective_intelligibility as stoi_tm,
+)
+
+import speechbrain as sb
+from speechbrain.dataio.dataio import write_audio
+from speechbrain.utils.metric_stats import MetricStats
+
+
+class SGMSEBrain(sb.Brain):
+    """
+    A Brain class to train an SGMSE-based diffusion model.
+    """
+
+    def on_fit_start(self):
+        """
+        Called once in the beginning of training.
+        """
+        super().on_fit_start()
+
+        log_dir = os.path.join(
+            self.hparams.tensorboard_logs, self.hparams.run_name
+        )
+        os.makedirs(log_dir, exist_ok=True)
+        self.writer = SummaryWriter(log_dir=log_dir)
+
+        ema = self.modules["score_model"].ema
+        self.checkpointer.add_recoverable(
+            name="ema",
+            obj=ema,
+            custom_save_hook=lambda obj, path: torch.save(
+                obj.state_dict(), path
+            ),
+            custom_load_hook=lambda obj, path, end: obj.load_state_dict(
+                torch.load(path)
+            ),
+        )
```

Comment:
This is fine, but you could also add the save/load code to the model itself with `@mark_as_saver` and `@mark_as_loader`

https://speechbrain.readthedocs.io/en/latest/API/speechbrain.utils.checkpoints.html#speechbrain.utils.checkpoints.register_checkpoint_hooks